src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPool.cs

```diff
@@ -655,34 +571,25 @@ private async ValueTask<HttpConnectionBase> GetHttp2ConnectionAsync(HttpRequestM
                 bool canUse = true;
                 lock (SyncObj)
                 {
+                    // Server does not support HTTP2. Disable further HTTP2 attempts.
                     _http2Enabled = false;
 
-                    if (request.Version.Major >= 2 && request.VersionPolicy != HttpVersionPolicy.RequestVersionOrLower)
-                    {
-                        sslStream.Close();
-                        throw new HttpRequestException(SR.Format(SR.net_http_requested_version_server_refused, request.Version, request.VersionPolicy));
-                    }
-
                     if (_associatedConnectionCount < _maxConnections)
                     {
                         IncrementConnectionCountNoLock();
                     }
                     else
                     {
-                        // We are in the weird situation of having established a new HTTP 1.1 connection
-                        // when we were already at the maximum for HTTP 1.1 connections.
-                        // Just discard this connection and get another one from the pool.
-                        // This should be a really rare situation to get into, since it would require
-                        // the user to make multiple HTTP 1.1-only requests first before attempting an
-                        // HTTP2 request, and the server failing to accept HTTP2.
+                        // We are already at the limit for HTTP/1.1 connections, so do not proceed with this connection.
                         canUse = false;
                     }
                 }
 
                 if (canUse)
                 {
-                    return await ConstructHttp11ConnectionAsync(async, socket, stream!, transportContext, request, cancellationToken).ConfigureAwait(false);
-               }
+                    HttpConnection http11Connection = await ConstructHttp11ConnectionAsync(async, socket, stream!, transportContext, request, cancellationToken).ConfigureAwait(false);
+                    ReturnConnection(http11Connection);
```

Comment:
So we create H/11 connection and immediately return it to the pool. So if I get it right, under certain circumstances, another request can take this connection in the meantime and the currently processed request will get queued instead? And that's what we want, or not, or we do not care?

Reply:
Yes, that's possible.

It's a weird scenario because you would have to be issuing both HTTP/1.1 and HTTP/2 requests against the same server (otherwise you wouldn't have any existing HTTP/1.1 requests queued). So I'm not sure we can that much...

But if you do that, and you do have HTTP/1.1 requests queued already, I'd argue that those queued requests should be serviced first. That's what will happen for any other HTTP2 requests that are implicitly queued by the HTTP2 connection creation semaphore. So I'd argue this behavior is both better and more consistent than existing behavior.